### PR TITLE
build wix v4 with Visual Studio Community 2017 only

### PR DESCRIPTION
**Pull request by andorz**
Had to change the OneTimeWixBuildInitialization.proj because the $(NetfxSdkInstallPath) and $(StrongNameSignFolder) do not work when using only Visual Studio Community 2017 to build.

The other changes belong to SHFB and dtfguide.helpproj. I had to use the newest nuget package and add parts for .Net 4.6.2.

### CODE REVIEW

File: tools/Nuget.targets [hunk 1]
```diff
@@ -14,7 +14,8 @@
     <ExtensionPackPath>$(NugetPackageFolder)MSBuild.Extension.Pack.1.2.0\lib\net40\</ExtensionPackPath>
     <LibGit2SharpPath>$(NugetPackageFolder)LibGit2Sharp.0.21.0.176\lib\net40\</LibGit2SharpPath>
     <LibGit2SharpPropsFilePath>$(NugetPackageFolder)LibGit2Sharp.0.21.0.176\build\net40\LibGit2Sharp.props</LibGit2SharpPropsFilePath>
-    <SHFBROOT>$(NugetPackageFolder)EWSoftware.SHFB.2014.11.22.0\Tools\</SHFBROOT>
+    <SHFBROOT>$(NugetPackageFolder)EWSoftware.SHFB.2017.1.28.0\Tools\</SHFBROOT>
```
Comment (rseanhall) [inline tools/Nuget.targets:17]: We can't upgrade SHFB until we migrate away from the obsolete Prototype style.

File: src/DTF/Documents/Reference/dtfref.shfbproj [hunk 2]
```diff
@@ -20,10 +20,13 @@
     <FooterText>&amp;lt%3bscript src=&amp;quot%3bhelplink.js&amp;quot%3b&amp;gt%3b&amp;lt%3b/script&amp;gt%3b</FooterText>
     <PresentationStyle>Prototype</PresentationStyle>
     <NamingMethod>MemberName</NamingMethod>
+	<FrameworkVersion>.NET Framework 4.6.2</FrameworkVersion>
```
Comment (rseanhall) [inline src/DTF/Documents/Reference/dtfref.shfbproj:23]: We aren't targeting 4.6.2, we're targeting 3.5 IIRC. Are you able to build with the old SHFB if you set this to 3.5?

**Comment by andorz** [comment]
When I use <FrameworkVersion>.NET Framework 3.5</FrameworkVersion> I can build but with warnings about some missing topics:

```
BuildAssembler : warning : ResolveReferenceLinksComponent: [M:WixToolset.Dtf.Compression.ArchiveFileStreamContext.GetArchiveName(System.Int32)] Unknown reference link target &#39;F:WixToolset.Dtf.Compression.ArchiveFileStreamContext.archiveFiles&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ShowMissingComponent: [M:WixToolset.Dtf.WindowsInstaller.CustomActionData.AddObject``1(System.String,``0)] Missing &lt;typeparam name=&quot;T&quot;/&gt; documentation [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ShowMissingComponent: [M:WixToolset.Dtf.WindowsInstaller.CustomActionData.GetObject``1(System.String)] Missing &lt;typeparam name=&quot;T&quot;/&gt; documentation [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.Components] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.Component_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.CreateFolders] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.CreateFolder_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.CustomActions] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.CustomAction_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.Directories] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.Directory_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.DuplicateFiles] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.DuplicateFile_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.FeatureComponents] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.FeatureComponent_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.Features] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.Feature_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.FileHashes] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.FileHash_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.Files] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.File_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.InstallExecuteSequences] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.InstallSequence_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.InstallUISequences] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.InstallSequence_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.LaunchConditions] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.LaunchCondition_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.Medias] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.Media_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.Properties] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.Property_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.Registries] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.Registry_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
BuildAssembler : warning : ResolveReferenceLinksComponent: [P:WixToolset.Dtf.WindowsInstaller.Linq.QDatabase.RemoveFiles] Unknown reference link target &#39;T:WixToolset.Dtf.WindowsInstaller.Linq.Entities.RemoveFile_&#39;. [D:\github\wix4\build\obj\ship\x86\dtfref\BuildReferenceTopics.proj]
```

**Comment by rseanhall** [comment]
I normally get warnings when building the DTF help in wix3 and wix4. It sounds like this fixed your issue without having to upgrade SHFB? Can you update the pull request?

**Pull request closed without merging**